### PR TITLE
OCSP Lookup Utility

### PR DESCRIPTION
**Pull request by cpu**
This PR adds a new binary `stapled-checkcert` to the project. This tool serves to perform one-off OCSP requests using the stapled codebase.

The intention is to provide an alternative to `openssl ocsp`, a command line utility used for fetching OCSP responses using OpenSSL. 

The OpenSSL 'utilities' are not meant for production usage, and are written in precarious C code. By comparison `stapled-checkcert` is Go based, requires no OpenSSL dep, and lets us exercise the common code already in place for stapled. In this way it can act as a "dry-run" check for certificates prior to placing them in the monitored directory. It is basically a re-implementation of @jsha's [ocsp.go](https://github.com/jsha/go/blob/master/ocsp/ocsp.go) utility but within the stapled codebase. I had a similar utility in my `ocspd` codebase.

Example invocation:

```
$> stapled-checkcert -h
Usage of stapled-checkcert:
  -issuer string
        PEM/DER encoded issuer file
  -timeout duration
        Max # of seconds before a request times out (default 5s)

$> stapled-checkcert ./test.pem 
16:29:18        main Fetching issuer from http://secure.globalsign.com/cacert/gsorganizationvalsha2g2r1.crt
16:29:18        main [fetcher] Sending request to 'http://ocsp2.globalsign.com/gsorganizationvalsha2g2/MFMwUTBPME0wSzAJBgUrDgMCGgUABBQMnk2cPe3vhNiR6XLHz4QGvBl7BwQUlt5h8b0cFilTHMDMfTuDAEDmGnwCEhEhl2tdXeumaJg3%2FvNgVRo8yw%3D%3D'
Cert: test
Good response:
  Status 0
  SerialNumber 1121976b5d5deba6689837fef360551a3ccb
  ProducedAt 2016-03-21 23:23:59 +0000 UTC
  ThisUpdate 2016-03-25 23:23:59 +0000 UTC
  NextUpdate 2016-03-25 23:23:59 +0000 UTC
  RevokedAt 0001-01-01 00:00:00 +0000 UTC
  RevocationReason 0J
  SignatureAlgorithm SHA256-RSA
  Extensions []pkix.Extension(nil)
```

This PR is based on a branch that assumes #28 is merged first.

### CODE REVIEW

File: stapled-checkcert/main.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	}
+
+	err = stapledOCSP.VerifyResponse(e.clk.Now(), e.serial, resp)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Cert: %s\n", e.name)
+	fmt.Printf("Good response:\n")
+	fmt.Printf("  Status %d\n", resp.Status)
```
Comment (rolandshoemaker) [inline stapled-checkcert/main.go:137]: It'd be nice to have a map of status -> string and revocation reason -> string (below) to make this a bit more readable, I thought `ocsp` actually had those itself but it appears not :/

**Comment by rolandshoemaker** [comment]
This looks great! I'll try to get my fixes in on #28 soon so we can unblock this.

**Comment by jsha** [comment]
Overall comment: Is it possible to reuse more of the stapled codebase for this? It feels like there's some unnecessary repetition, but I haven't taken a very close look yet.

**Comment by cpu** [comment]
> Is it possible to reuse more of the stapled codebase for this?

Likely! I'll give it a go.

**Comment by cpu** [comment]
I long ago forgot about this! I'm going to close this PR and maybe revisit from scratch one day if its still a thing someone wants.

**Pull request closed without merging**